cirq-core/cirq/sim/simulator_base_test.py

```diff
@@ -194,6 +187,7 @@ def test_simulate_empty_circuit():
     sim = CountingSimulator()
     r = sim.simulate(cirq.Circuit())
     assert r._final_simulator_state.gate_count == 0
+    assert r._final_simulator_state.measurement_count == 0
```

Comment:
Similar to my comment above, this would be inconsistent with existing behavior. Simulator state is (and has always been) passed by reference into the step results, for performance reasons. Users who need a durable copy of the state for each step need to explicitly copy that state immediately after each step result.